client/ayon_shotgrid/plugins/publish/validate_shotgrid_user.py

```diff
@@ -0,0 +1,48 @@
+import pyblish.api
+from openpype.pipeline.publish import ValidateContentsOrder
+
+
+class ValidateShotgridUser(pyblish.api.ContextPlugin):
+    """
+    Check if user is valid and have access to the project.
+    """
+    label = "Validate Shotgrid User"
+    order = ValidateContentsOrder
+
+    def process(self, context):
+        sg_session = context.data.get("shotgridSession")
+        user_login = context.data.get("shotgridUser")
+        sg_project = context.data.get("shotgridProject")
+        project_name = context.data["projectEntity"]["name"]
+
+        if not (user_login and sg_session and sg_project):
+            raise KeyError()
```

Comment:
This would require import `from openpype.pipeline import PublishValidationError`
```suggestion
            raise PublishValidationError("Missing shotgrid credentials")
```